src/otp-input/otp-input.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useEffect, useRef, useState } from "react";
+import { OtpInputProps } from "./types";
+import {
+    CTAButton,
+    ErrorContainer,
+    InputContainer,
+    InputField,
+    Wrapper,
+} from "./otp-input.styles";
+
+export const OtpInput = ({
+    cooldownDuration,
+    actionButtonProps,
+    errorMessage,
+    numOfInput,
+    onChange,
+    ...otherProps
+}: OtpInputProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const {
+        disabled,
+        onClick,
+        styleType = "secondary",
+        ...otherCtaProps
+    } = actionButtonProps ?? {};
+    const inputRefs = useRef<(HTMLInputElement | null)[]>([]);
+
+    const [otpValues, setOtpValues] = useState<string[]>(
+        new Array(numOfInput).fill("")
+    );
+    const [countDown, setCountDown] = useState<number>(cooldownDuration);
```

Comment:
I would recommend using a timestamp to track the start of the countdown. Chrome may delay the timer in inactive tabs, resulting in desynced countdowns

Reply:
Amended to use the timestamp. The countdown is just for cosmetic display